hazelcast-jet-core/src/main/java/com/hazelcast/jet/Observable.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet;
+
+import javax.annotation.Nonnull;
+import java.util.function.Consumer;
+
+/**
+ * Represents a flowing sequence of values. The sequence can be observed
+ * by registering {@link Observer}s on it. Observers are able to see
+ * all the values that show up in the sequence after they have been
+ * registered.
+ * <p>
+ * Besides new values appearing observers can also observe completions and
+ * failure event. Completion means that no further values will appear in
+ * the sequence. Failure means that something went wrong during the
+ * production of the sequence's values and the event attempts to provide
+ * useful information about the cause of the problem.
+ *
```

Comment:
We should describe the lifecycle in more detail. We should mention that it's a standalone named object that needs to be destroyed and that if observable is used in job, it will be cleaned up one day.
We should also describe what happens if we reuse the name for further jobs or if we use the same observable for multiple sinks in the job.